src/UserAction/StepExecutorTrait.php

```diff
@@ -102,18 +105,53 @@ protected function setFormField(Form $form, array $fields, string $step): Form
         foreach ($fields as $k => $val) {
             $form->getControl($k)->set($val);
         }
+
         $this->hook(self::HOOK_STEP, [$step, $form]);
 
         return $form;
     }
 
+    /**
+     * Set model for userAction arguments.
+     * Override existing argument with model definition.
+     */
+    protected function getModelForArgs(): Model
+    {
+        $args = $this->getAction()->args;
+        if (array_key_exists('__atk_model', $args)) {
+            /** @var Model $argsModel */
+            $argsModel = Factory::factory($args['__atk_model']);
+            // if seed is supplied, we need to initialize.
+            if (!$argsModel->_initialized) {
+                $argsModel->invokeInit();
```

Comment:
Model::init is invoked when persistence is set

when not initialized, I belive we should set `new Array_([])` persistence instead

side question, when custom model for args does bring any advantage?

Reply:
- argument can be easily validate using a Validator;
- argument can be easily define within a model;